core/src/idx/ft/analyzer/mapper.rs

```diff
@@ -1,41 +1,72 @@
 use crate::err::Error;
 use crate::idx::ft::analyzer::filter::{FilterResult, Term};
+#[cfg(target_arch = "wasm32")]
+use std::fs::File;
+#[cfg(target_arch = "wasm32")]
+use std::io::{BufRead, BufReader};
 use std::path::Path;
 use std::str::FromStr;
 use std::sync::Arc;
+#[cfg(not(target_arch = "wasm32"))]
 use tokio::fs::File;
+#[cfg(not(target_arch = "wasm32"))]
 use tokio::io::{AsyncBufReadExt, BufReader};
 use vart::art::Tree;
 use vart::VariableSizeKey;
-
 #[derive(Clone, Default)]
 pub(in crate::idx) struct Mapper {
 	terms: Arc<Tree<VariableSizeKey, String>>,
 }
 
 impl Mapper {
 	pub(in crate::idx) async fn new(path: &Path) -> Result<Self, Error> {
+		let mut terms = Tree::new();
+		Self::iterate_file(&mut terms, path).await?;
+		Ok(Self {
+			terms: Arc::new(terms),
+		})
+	}
+
+	fn add_line_tree(terms: &mut Tree<VariableSizeKey, String>, line: String) -> Result<(), Error> {
+		let mut parts = line.splitn(2, '\t');
+		if let Some(lemme) = parts.next() {
+			if let Some(word) = parts.next() {
+				let key = VariableSizeKey::from_str(word)
+					.map_err(|_| Error::Internal(format!("Can't create key from {word}")))?;
+				terms
+					.insert(&key, lemme.to_string(), 0, 0)
```

Comment:
If this is not going to be accessed concurrently, insert_unchecked should give much better perf because it avoids CoW, whereas insert internally uses CoW approach